beacon_node/network/src/sync/block_lookups/single_block_lookup.rs

```diff
@@ -319,12 +322,42 @@ impl<L: Lookup, T: BeaconChainTypes> SingleBlockLookup<L, T> {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct RequestedBlobIds(Vec<BlobIdentifier>);
+
+impl From<Vec<BlobIdentifier>> for RequestedBlobIds {
+    fn from(value: Vec<BlobIdentifier>) -> Self {
+        Self(value)
+    }
+}
+
+impl Into<VariableList<BlobIdentifier, MaxRequestBlobSidecars>> for RequestedBlobIds {
+    fn into(self) -> VariableList<BlobIdentifier, MaxRequestBlobSidecars> {
+        VariableList::from(self.0)
+    }
+}
+
+impl RequestedBlobIds {
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+    pub fn contains(&self, blob_id: &BlobIdentifier) -> bool {
+        self.0.contains(blob_id)
+    }
+    pub fn remove(&mut self, blob_id: &BlobIdentifier) {
+        self.0.retain(|id| id != blob_id)
```

Comment:
can just be a remove no?

Reply:
I think the `Vec<BlobIdentifier>` isn't *necessarily* always orded by blob index, but that's a good point, let me see if I can add a little refactor to make sure that's always the case

Reply:
Oh I think the issues that we might get any blob index in a response and when we remove it, `remove` shifts all the elements, so trying to access the again by blob index wouldn't work

Reply:
Ohh sorry thought it was a hashset for some reason